lib/galaxy/jobs/__init__.py

```diff
@@ -1375,11 +1375,13 @@ def path_rewriter(path):
                             return path
 
                         dataset.metadata.from_JSON_dict(output_filename, path_rewriter=path_rewriter)
+                    line_count = context.get('line_count', None)
                     try:
-                        assert context.get('line_count', None) is not None
-                        dataset.set_peek(line_count=context['line_count'])
-                    except Exception:
-                        dataset.set_peek()
+                        # Certain datatype's set_peek methods contain a line_count argument
+                        dataset_assoc.dataset.set_peek(line_count=line_count)
+                    except TypeError:
```

Comment:
Would it be worth forcing a standard interface for set_peek down the road?  Just food for thought -- not sure I have a strong opinion one way or the other, but that'd be an alternative to having to catch and re-invoke here.

Reply:
absolutely, I think we should do that.